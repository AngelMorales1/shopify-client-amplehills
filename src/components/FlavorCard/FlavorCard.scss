@import 'settings', 'typography';

.FlavorCard {
  border-radius: radius('xs');
  width: spacing(65);
  height: spacing(90);

  @media ($medium-up) {
    width: spacing(75);
    height: spacing(100);
  }

  &__mark {
    width: spacing(20);
    height: spacing(20);
  }

  &__mark-text {
    width: spacing(20);
    font-size: $font-2;
    letter-spacing: spacing(0.1);
  }

  &__image {
    height: 80%;
  }
}
