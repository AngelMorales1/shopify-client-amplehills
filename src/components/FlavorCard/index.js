import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import contentfulImgUtil from 'utils/contentfulImgUtil';

import { Button } from 'components/base';
import styles from './FlavorCard.scss';

const FlavorCard = ({ flavor }) => {
  return (
    <Button variant="style-none" to={`/flavors/${flavor.slug}`}>
      <div
        className={cx(
          styles['FlavorCard'],
          'bg-white m2 p3 flex flex-column justify-between items-center relative'
        )}
      >
        {flavor.label ? (
          <div
            className={cx(
              styles['FlavorCard__mark'],
              'z-subnav circle bg-peach absolute z-1 flex items-center justify-center m2 r0 t0 p1',
              {
                'bg-pastel-blue': flavor.labelColor === 'blue',
                'bg-turquoise': flavor.labelColor === 'green'
              }
            )}
          >
            <p
              className={cx(
                styles['FlavorCard__mark-text'],
                'light center carter text-white white-space-normal'
              )}
            >
              {flavor.label}
            </p>
          </div>
        ) : null}
        <div
          className={cx(styles['FlavorCard__image'], 'col-12')}
          style={{
            background: `url(${contentfulImgUtil(
              flavor.image,
              '200'
            )}) no-repeat center`,
            backgroundSize: 'contain'
          }}
        />
        <p className="center bold text-madison-blue">{flavor.title}</p>
      </div>
    </Button>
  );
};

FlavorCard.propTypes = {
  flavor: PropTypes.shape({
    dietaryRestrictions: PropTypes.object,
    filters: PropTypes.object,
    id: PropTypes.string,
    image: PropTypes.string,
    label: PropTypes.string,
    title: PropTypes.string
  })
};

FlavorCard.defaultProps = {
  flavor: {
    dietaryRestrictions: {},
    filters: {},
    id: '',
    image: '',
    label: '',
    title: ''
  }
};

export default FlavorCard;
