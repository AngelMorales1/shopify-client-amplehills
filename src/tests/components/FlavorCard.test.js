import React from 'react';
import { shallow } from 'enzyme';

import FlavorCard from 'components/FlavorCard';

it('renders without data', () => {
  const component = shallow(<FlavorCard />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <FlavorCard
      flavors={{
        dietaryRestrictions: {},
        filters: {},
        id: '',
        image: '',
        label: '',
        title: ''
      }}
    />
  );

  expect(component).toMatchSnapshot();
});
