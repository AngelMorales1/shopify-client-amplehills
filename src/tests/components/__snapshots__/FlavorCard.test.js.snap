// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlavorCard
    flavor={
        Object {
            "dietaryRestrictions": Object {},
            "filters": Object {},
            "id": "",
            "image": "",
            "label": "",
            "title": "",
          }
    }
    flavors={
        Object {
            "dietaryRestrictions": Object {},
            "filters": Object {},
            "id": "",
            "image": "",
            "label": "",
            "title": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="bg-white m2 p3 flex flex-column justify-between items-center relative"
>
        <div
                className="col-12"
                style={
                        Object {
                                "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                                "backgroundSize": "contain",
                              }
                }
        />
        <p
                className="center bold text-madison-blue"
        />
</div>,
      "className": "",
      "color": "white-denim",
      "disabled": false,
      "fullWidth": false,
      "hover": "shadow",
      "label": "",
      "onClick": [Function],
      "shadow": false,
      "to": "/flavors/",
      "type": "button",
      "variant": "style-none",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="col-12"
            style={
                        Object {
                                    "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                                    "backgroundSize": "contain",
                                  }
            }
/>,
          <p
            className="center bold text-madison-blue"
/>,
        ],
        "className": "bg-white m2 p3 flex flex-column justify-between items-center relative",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-12",
            "style": Object {
              "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
              "backgroundSize": "contain",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "center bold text-madison-blue",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="bg-white m2 p3 flex flex-column justify-between items-center relative"
>
          <div
                    className="col-12"
                    style={
                              Object {
                                        "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                                        "backgroundSize": "contain",
                                      }
                    }
          />
          <p
                    className="center bold text-madison-blue"
          />
</div>,
        "className": "",
        "color": "white-denim",
        "disabled": false,
        "fullWidth": false,
        "hover": "shadow",
        "label": "",
        "onClick": [Function],
        "shadow": false,
        "to": "/flavors/",
        "type": "button",
        "variant": "style-none",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <div
              className="col-12"
              style={
                            Object {
                                          "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                                          "backgroundSize": "contain",
                                        }
              }
/>,
            <p
              className="center bold text-madison-blue"
/>,
          ],
          "className": "bg-white m2 p3 flex flex-column justify-between items-center relative",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "col-12",
              "style": Object {
                "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                "backgroundSize": "contain",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "center bold text-madison-blue",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlavorCard
    flavor={
        Object {
            "dietaryRestrictions": Object {},
            "filters": Object {},
            "id": "",
            "image": "",
            "label": "",
            "title": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="bg-white m2 p3 flex flex-column justify-between items-center relative"
>
        <div
                className="col-12"
                style={
                        Object {
                                "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                                "backgroundSize": "contain",
                              }
                }
        />
        <p
                className="center bold text-madison-blue"
        />
</div>,
      "className": "",
      "color": "white-denim",
      "disabled": false,
      "fullWidth": false,
      "hover": "shadow",
      "label": "",
      "onClick": [Function],
      "shadow": false,
      "to": "/flavors/",
      "type": "button",
      "variant": "style-none",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="col-12"
            style={
                        Object {
                                    "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                                    "backgroundSize": "contain",
                                  }
            }
/>,
          <p
            className="center bold text-madison-blue"
/>,
        ],
        "className": "bg-white m2 p3 flex flex-column justify-between items-center relative",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-12",
            "style": Object {
              "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
              "backgroundSize": "contain",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "center bold text-madison-blue",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="bg-white m2 p3 flex flex-column justify-between items-center relative"
>
          <div
                    className="col-12"
                    style={
                              Object {
                                        "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                                        "backgroundSize": "contain",
                                      }
                    }
          />
          <p
                    className="center bold text-madison-blue"
          />
</div>,
        "className": "",
        "color": "white-denim",
        "disabled": false,
        "fullWidth": false,
        "hover": "shadow",
        "label": "",
        "onClick": [Function],
        "shadow": false,
        "to": "/flavors/",
        "type": "button",
        "variant": "style-none",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <div
              className="col-12"
              style={
                            Object {
                                          "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                                          "backgroundSize": "contain",
                                        }
              }
/>,
            <p
              className="center bold text-madison-blue"
/>,
          ],
          "className": "bg-white m2 p3 flex flex-column justify-between items-center relative",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "col-12",
              "style": Object {
                "background": "url(?w=200&fm=jpg&fl=progressive&q=90) no-repeat center",
                "backgroundSize": "contain",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "center bold text-madison-blue",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
